src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -926,6 +948,28 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 						reflectionContext.RecordUnrecognizedPattern ($"Activator call '{reflectionContext.MethodCalled.FullName}' inside '{reflectionContext.MethodCalling.FullName}' is not yet supported");
 						break;
 
+					//
+					// System.Runtime.CompilerServices.RuntimeHelpers
+					//
+					// RunClassConstructor (RuntimeTypeHandle type)
+					//
+					case IntrinsicId.RuntimeHelpers_RunClassConstructor: {
+							reflectionContext.AnalyzingPattern ();
+							foreach (var TypeHandleValue in methodParams [0].UniqueValues ()) {
+								if (TypeHandleValue is RuntimeTypeHandleValue runtimeTypeHandleValue) {
+									_markStep.MarkStaticConstructor (runtimeTypeHandleValue.TypeRepresented, new DependencyInfo (DependencyKind.AccessedViaReflection, reflectionContext.MethodCalling));
+									reflectionContext.RecordHandledPattern ();
+								} else if (TypeHandleValue == NullValue.Instance)
+									reflectionContext.RecordHandledPattern ();
+								else {
+									reflectionContext.RecordUnrecognizedPattern ($"A {GetValueDescriptionForErrorMessage (TypeHandleValue)} " +
+										$"is passed into the {GetMetadataTokenDescriptionForErrorMessage (reflectionContext.MethodCalling)}. " +
+										$"It's not possible to guarantee that these requirements are met by the application.");
```

Comment:
Maybe we should be a bit more specific - just a suggestion: "It's not possible to guarantee availability of the target static constructor." or something like that.